lib/nfg_ui/components/elements/button.rb

```diff
@@ -76,10 +78,14 @@ def left_icon_component
 
         def right_icon_component
           NfgUi::Components::Foundations::Icon.new({ traits: [icon, :right],
-                                                     class: NfgUi::Components::Foundations::Icon::RIGHT_ICON_SPACER_CSS_CLASS },
+                                                     class: right_icon_class },
                                                    view_context).render
         end
 
+        def right_icon_class
+          NfgUi::Components::Foundations::Icon::RIGHT_ICON_SPACER_CSS_CLASS if body.present?
+        end
```

Comment:
Since most ruby changes are worth noting, this one doesn't need to be noted in the docs -- it was a bug fix -- it removes extra space on buttons when an icon is present but there's no text.